File: java/program/Shirt.java
// deklarasi class dan menjadi anak Product
class Shirt extends Clothing {

    // attibut private bertipe data string
    private String color;
    private String sleeve_type;

    // constructor tanpa parameter
    public Shirt() {
        this.color = "";
        this.sleeve_type = "";

    }

    // constructor dengan parameter color, sleeve_type
    public Shirt(String color, String sleeve_type) {
        this.color = color;
        this.sleeve_type = sleeve_type;
    }

    // setter dan getter untuk setiap atribut color, sleevetype
    public String get_color() {
        return this.color;
    }

    public void set_color(String color) {
        this.color = color;
    }

    public String get_sleeve_type() {
        return this.sleeve_type;
    }

    public void set_sleeve_type(String sleeve_type) {
        this.sleeve_type = sleeve_type;
    }

};

File: c++/program/Shirt.cpp
// import library
#include <iostream>
#include <string>
#include "Clothing.cpp"

using namespace std;

// deklarasi class menjadi anak dari Clothing
class Shirt : public Clothing
{

    // attibut private bertipe data string
private:
    string color;
    string sleeve_type;

    // public Methods
public:
    // constructor tanpa parameter
    Shirt()
    {
        this->color = "";
        this->sleeve_type = "";
    }

    // constructor
    Shirt(string color, string sleeve_type)
    {
        this->color = color;
        this->sleeve_type = sleeve_type;
    }

    // setter dan getter
    string get_color()
    {
        return this->color;
    }

    void set_color(string color)
    {
        this->color = color;
    }

    string get_sleeve_type()
    {
        return this->sleeve_type;
    }

    void set_sleeve_type(string sleeve_type)
    {
        this->sleeve_type = sleeve_type;
    }

    // deconstructor
    ~Shirt()
    {
    }
};
